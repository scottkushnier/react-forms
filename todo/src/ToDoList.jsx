
import { useState } from 'react'
import ToDo from './ToDo.jsx'
import NewToDo from './NewToDo.jsx';

let keyInd = 1;
function newKey() {
    const res = keyInd;
    keyInd++;
    return res;
}

// const initialList = [{text: "Count the bees", key: newKey()},
//   {text: "Change tablecloth", key: newKey()},
//   {text: "Wash beard", key: newKey()}];

const initialList = [];

function ToDoList() {
    const removeItemWithKey = key => setToDos(toDos.filter(box => (box.key != key)));
    const addToDo = (formData) => {
      const {text} = formData;
      setToDos([...toDos, {text, key: newKey()}]);
    }
    const [toDos, setToDos] = useState(initialList);
    return (
        <div data-testid="todolist">
          <h2> Checklist </h2>
          {toDos.map(({text, key}) => {
               return <ToDo text={text} key={key}
                            deleteFn={() => {removeItemWithKey(key)}}
                      />
          })}
          <hr/>
          <NewToDo AddFn={addToDo}/>
        </div>
    )
}

export default ToDoList;