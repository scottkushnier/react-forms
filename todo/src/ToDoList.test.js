import { render, fireEvent } from "@testing-library/react";
import ToDoList from "./ToDoList";
import '@testing-library/jest-dom'

test("todolist smoke test", function () {
  render(<ToDoList />)
});

test("matches snapshot", function() {
    const {asFragment} = render(<ToDoList />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("should add new todo item", function() {
    const {getByLabelText, queryByTestId} = render(<ToDoList />);
    const itemInput = getByLabelText('New Item:');
    const div = queryByTestId('todolist');
    const form = queryByTestId('form')
    expect(div).not.toContainHTML("shopping");
    fireEvent.change(itemInput, {target: {value: 'shopping'}});
    fireEvent.submit(form);
    expect(div).toContainHTML("shopping");
});
