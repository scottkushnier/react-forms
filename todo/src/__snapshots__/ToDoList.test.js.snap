// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="todolist"
  >
    <h2>
       Checklist 
    </h2>
    <hr />
    <form
      data-testid="form"
    >
      <br />
      <label
        for="item"
      >
        New Item: 
      </label>
      <input
        id="item"
        name="text"
        value=""
      />
    </form>
  </div>
</DocumentFragment>
`;
